integrationutils.go

```diff
@@ -102,6 +120,103 @@ func getOpenPullRequests(t *testing.T, client vcsclient.VcsClient, testDetails *
+	defer removeBranchFunc()
+
+	ctx := context.Background()
+	// Create a pull request from the timestamp based issue branch against the main branch
+	err := client.CreatePullRequest(ctx, repoOwner, repoName, currentIssuesBranch, mainBranch, "scan pull request integration test", "")
+	require.NoError(t, err)
+
+	// Find the relevant pull request id
+	pullRequests := getOpenPullRequests(t, client, testDetails)
+	prId := findRelevantPrID(pullRequests, currentIssuesBranch)
+	testDetails.PullRequestID = strconv.Itoa(prId)
+	require.NotZero(t, prId)
+	defer func() {
+		err = client.UpdatePullRequest(ctx, repoOwner, repoName, "scan pr test finished", "", "", prId, vcsutils.Closed)
+		assert.NoError(t, err)
+	}()
+
+	// Set the required environment variables for the scan-pull-request command
+	unsetEnvs := setIntegrationTestEnvs(t, testDetails)
+	defer unsetEnvs()
+
+	err = Exec(&scanpullrequest.ScanPullRequestCmd{}, utils.ScanPullRequest)
+	// Validate that issues were found and the relevant error returned
+	require.Errorf(t, err, scanpullrequest.SecurityIssueFoundErr)
+
+	validateResults(t, ctx, client, prId)
+}
+
+func runScanRepositoryCmd(t *testing.T, client vcsclient.VcsClient, testDetails *IntegrationTestDetails) {
+	_, restoreFunc := utils.ChangeToTempDirWithCallback(t)
+	defer func() {
+		assert.NoError(t, restoreFunc())
+	}()
+
+	timestamp := getTimestamp()
+	// Add a timestamp to the fixing pull requests, to identify them later
+	testDetails.CustomBranchName = "frogbot-{IMPACTED_PACKAGE}-{BRANCH_NAME_HASH}-" + timestamp
+
+	// Set the required environment variables for the scan-repository command
+	unsetEnvs := setIntegrationTestEnvs(t, testDetails)
+	defer unsetEnvs()
+
+	err := Exec(&scanrepository.ScanRepositoryCmd{}, utils.ScanRepository)
+	assert.NoError(t, err)
+
+	gitManager := buildGitManager(t, testDetails)
+
+	pullRequests := getOpenPullRequests(t, client, testDetails)
+
+	expectedBranchName := "frogbot-pyjwt-45ebb5a61916a91ae7c1e3ff7ffb6112-" + timestamp
+	assert.NoError(t, gitManager.RemoveRemoteBranch(expectedBranchName))
+	prId := findRelevantPrID(pullRequests, expectedBranchName)
+	assert.NotZero(t, prId)
+	ctx := context.Background()
+	err = client.UpdatePullRequest(ctx, repoOwner, repoName, "scan repository test finished", "", "", prId, vcsutils.Closed)
+	assert.NoError(t, err)
+
+	expectedBranchName = "frogbot-pyyaml-985622f4dbf3a64873b6b8440288e005-" + timestamp
+	prId = findRelevantPrID(pullRequests, expectedBranchName)
+	assert.NoError(t, gitManager.RemoveRemoteBranch(expectedBranchName))
+	assert.NotZero(t, prId)
+	err = client.UpdatePullRequest(ctx, repoOwner, repoName, "scan repository test finished", "", "", prId, vcsutils.Closed)
+	assert.NoError(t, err)
+}
+
+func validateResults(t *testing.T, ctx context.Context, client vcsclient.VcsClient, prId int) {
+	switch client.(type) {
+	case *vcsclient.GitHubClient:
+		comments, err := client.ListPullRequestComments(ctx, repoOwner, repoName, prId)
+		assert.NoError(t, err)
```

Comment:
Let's move these lines above the switch to reuse them in vcsclient.AzureReposClient too